soc/arm/st_stm32/stm32h7/Kconfig.defconfig.stm32h7a3xx

```diff
@@ -6,7 +6,8 @@
 if SOC_STM32H7A3XX
 
 config SOC
-	default "stm32h7a3xx"
+	default "stm32h7a3xxQ" if POWER_SUPPLY_DIRECT_SMPS
+	default "stm32h7a3xx" if !POWER_SUPPLY_DIRECT_SMPS
```

Comment:
```suggestion
	default "stm32h7a3xxQ" if SOC_STM32H7A3XXQ
	default "stm32h7a3xx" if !SOC_STM32H7A3XXQ
```

Reply:
@erwango 
I made all the changes as your request, but please note we have a build warning consequently build error,

The reason is the way we select `SOC_STM32H7A3XX` from `SOC_STM32H7A3XXQ`. So what should we do?

```
warning: the choice symbol SOC_STM32H7A3XX (defined at soc/arm/st_stm32/stm32h7/Kconfig.soc:45) is selected by the following symbols, but select/imply has no effect on choice symbols
 - SOC_STM32H7A3XXQ (defined at soc/arm/st_stm32/stm32h7/Kconfig.soc:50)

error: Aborting due to Kconfig warnings
```

Reply:
@manoj153 Ok, thanks for having add a try. I also just made some attempts on my side, and I don't find any better way than finally defining 2 different, independent SoC symbols, `SOC_STM32H7A3XX` and `SOC_STM32H7A3XXQ` that will have to live separately.
So let's do this:

```
 config SOC_STM32H723XX
 	bool "STM32H7A3XX"
 	select CPU_CORTEX_M7
 	select CPU_HAS_FPU_DOUBLE_PRECISION
 
 config SOC_STM32H7A3XXQ
 	bool "STM32H7A3XXQ"
	select CPU_CORTEX_M7
	select CPU_HAS_FPU_DOUBLE_PRECISION
```

Then:
```
 if SOC_STM32H7A3XX || SOC_STM32H7A3XXQ
 
 config SOC
	default "stm32h7a3xxQ" if SOC_STM32H7A3XXQ
	default "stm32h7a3xx" if SOC_STM32H7A3XX
 
 config NUM_IRQS
 	default 155
 
 endif # SOC_STM32H7A3XX
```

And finally, use `defined(CONFIG_SOC_STM32H7A3XX) || defined(CONFIG_SOC_STM32H7A3XXQ)`
instead of `defined(CONFIG_SOC_STM32H7A3XX)`.

I hope we can do better once SMPS is moved to device tree.

Anyway, thanks for your patience in doing the multiple tries.

Reply:
Thanks, @erwango. I have made it as requested.